src/doom/r_data.c

```diff
@@ -1196,6 +1196,29 @@ void R_InitColormaps (void)
 	byte *const playpal = W_CacheLumpName("PLAYPAL", PU_STATIC);
 	byte *const colormap = W_CacheLumpName("COLORMAP", PU_STATIC);
 
+	// [crispy] Smoother diminishing lighting.
+	// Compiled in but not enabled TrueColor mode
+	// can't use more than original 32 colormaps.
+	if (crispy->truecolor && crispy->smoothlight)
+	{
+		NUMCOLORMAPS = 256;
+	}
+	else
+	{
+		NUMCOLORMAPS = 32;
+	}
+	// [crispy] Regardless of the number of colormaps,
+	// invulnerability is always the last one.
+	INVERSECOLORMAP = NUMCOLORMAPS;
```

Comment:
Why does this work at all since `INVERSECOLORMAP` has been turned into a macro?

Reply:
Oh god... It's a first time I tried to set something directly to macro. Interestingly, neither GCC nor Clang was giving a warning about it. Removed this block entirely, it's now macrocized.

Reply:
Well, after evaluating the macro, the line read `NUMCOLORMAPS = NUMCOLORMAPS;` which is a no-op that the compiler merely optimized out.

Reply:
Ah. I was thinking about macrocizing `INVERSECOLORMAP` yesterday, but mistaken cowardice prevented to do it - I was thinking that via macro `INVERSECOLORMAP` will be always uninitialized `0`, but this is wrong of course, this is not how macros are working and they can accept `int` and other changeable values.